### ISSUE TITLE
Introduce Sound Theme, similar like Product Icon Themes

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#145204]
Some users want more customisation towards what sounds are being used.
I think in the future we should consider introducing Sound Themes similar to Icon Themes. So they can be extensions and users can choose which one is applied.

For now assigning this to the backlog.
fyi @aeschli @hediet 

@gino-scarpino to try out the product icon experience do F1 > Preferences Product Icon Themes > Browse Additional Product Icon Themes...

**Comment by hediet** [comment]
Interesting idea.
Shouldn't we start with overridable audio cues first though? (the same as users can override specific theme colors)

**Comment by isidorn** [comment]
@hediet but the theme color you can specify inside the settings.json, for sounds it would have to be a path to the audio file which is in `opus` format. Which I find ugly. I just thought that the Theme would be the "best packaged" solution. Same as how we do not allow specific overrides for product icon themes.

**Comment by misolori** [comment]
This seems like a natural progression from our customization options and extension API 👍  

though I do imagine seeing this a bigger hurdle for extension authors as it requires some knowledge of generating/constructing audio sounds (similar to product icon themes requiring icon design). but from a principle, this makes a lot of sense.

**Comment by misolori** [comment]
@gino-scarpino essentially, yes. This is one of the differentiators of the product where you can customize nearly everything: color themes, settings, icon themes. While it may sound like a crazy idea at first, this really opens the door for others to push the boundaries and give users more options. I initially wasn't a fan of it for product icon themes but this has made it possible for others to find the right customizations that fit their own workflow (see [these product icon theme extensions](https://marketplace.visualstudio.com/search?term=tag%3Aproduct-icon-theme&target=VSCode&category=All%20categories&sortBy=Relevance) that have started within the last year).

**Comment by MarximusMaximus** [comment]
> This seems like a natural progression from our customization options and extension API 👍
> 
> though I do imagine seeing this a bigger hurdle for extension authors as it requires some knowledge of generating/constructing audio sounds (similar to product icon themes requiring icon design). but from a principle, this makes a lot of sense.

Could you elaborate/link any docs where I could get started on just overriding the default sounds? 

The breakpoint sound is making a weird clicking in my left ear and it's really annoying. It is more pronounced in the left channel, although it is still there in the right channel (swapped headphone ears to check). It is right at the end of the sound. Gets very pronounced if you're hitting breakpoint after breakpoint quickly.

**Comment by misolori** [comment]
> Could you elaborate/link any docs where I could get started on just overriding the default sounds?

This is not currently possible and is why this issue remains open on the backlog. If you are encountering any audio issues, please create a new issue for us to take a look at. Thanks!

**Comment by AFre100** [comment]
Hi @MarximusMaximus, 

cc @gino-scarpino 

I'm a sound designer working for Microsoft. Thanks for bringing this to our attention! Sounds like some friction -- this issue is not taken lightly.

I've included some potential fixes below and information asks from your end to help us zero in on this for yourself and other users. 


**Please download and test/play these (very short) audio [files ](https://www.dropbox.com/s/o36dh9kc0c20q7m/Clicking%20Playback%20Test%202.zip?dl=0
)in Media Player. The link will expire in the near future** 
Can you indicate the amount of “click” you hear in each, if any? 



**Troubleshooting**

 • I know that you know this, but just wanted to triple check that this problem persists after restarting your computer?

 • Hardware, out of date drivers, high CPU, and a few other culprits are common causes for popping and crackling in Windows 10 & 11. A few links below have troubleshooting exercises while will help you solve the issue or isolate it -- does it still persist? 

[The Geek Page](https://thegeekpage.com/audio-sound-popping-on-in-windows-10-solved/#Fix_6_-_Disable_fast_Startup
)
[Winbuzzer](https://winbuzzer.com/2022/02/18/how-to-fix-audio-popping-or-crackling-in-windows-10-xcxwbt/
)
[MS Community Forum
](https://answers.microsoft.com/en-us/windows/forum/all/audio-cracklingpopping-possible-issue-due-to/1e9bbf43-9eac-4596-9092-d680da4a71e6)

 • Which machine, OS, version of VS Code are you running while experiencing this issue?

 • Are headphones plugged in directly to the computer? If not, what hardware is it plugged into (e.g. display monitor, audio interface, HDMI device, etc)?

**If the problem is still persisting:**
 • Any audio of the "click" you can capture would be helpful. Can you please provide a screen grab or recording of it happening? If that's difficult to do, even a cell phone video recording would be better than nothing if it's audible.

**Comment by isidorn** [comment]
@MarximusMaximus friendly ping for this. Have you had a chance to try out the sounds @AFre100 provided.
Thanks!

**Comment by MarximusMaximus** [comment]
I'm starting to think it might be a local issue to my system, I've started having issues w/ FaceTime's & YouTube's audio now too where there is a complete silence either before or after speech. At the time, I thought maybe it was just the one VSCode sound file since I could reproduce it in VLC with that file and not another, but now it appears to be happening with other things. I haven't had time to isolate further if it's a system issue or just that my whole setup happens to amplify those hard silence cuts. But I haven't forgotten that this is on my list of TODO.

macOS 12.4 (21F79)
RogueAmoeba software (latest): SoundSource, AudioHijack, Loopback
NDI 5 & 4.x
OBS latest
NDI-obs latest
VSCode latest stable + various extensions

**Comment by isidorn** [comment]
@MarximusMaximus thanks. Do let us know when you have more details.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by sinewalker** [comment]
> Sorry, clarifying question, are we saying you'd like to light up the ability for the community to create and implement their own sounds? I know I'm new to the VS Code mindset, but this sounds like a crazy, out of control idea to me and I'd need to be convinced as to how, for sound, this would end in anything but cacophony. Thanks, G
> […](#)

Exactly, _we would like the ability to create (or even just borrow) our own sounds_.

Why do you think this a crazy, out of control idea? The current sounds are far from ideal, and [I am not alone in thinking that the one for Copilot is grating and annoying](https://stackoverflow.com/questions/73827540/annoying-sound-with-github-copilot-when-making-in-line-suggetions-in-visual-stud), so I switched it off.

Even if many "sound themes" for VSCode are made and shared in the community, no-one is forcing anyone to install any of them, so I don't see that there would be a cacophony.

**Comment by stellarpower** [comment]
Sound themes make sense, but I'd still like to be able just to provide my own path. Most linux desktops have different sound packs, but I can generally go into the settings and just pick a local file if I prefer that. Same for phone; I'm not limited to the ringtones provided even if these suit most users. If I'm on the train and hear the Takeshi's Castle theme tune playing, I sure as hell know that's my phone and not someone else's; if I have a custom ringtone I know who's calling before I pick up.

Most of the use for audio cues would obviously be for assistive technologies; in my case, I want the sound because my code may take a long time to run, and I want a loud and potentially very annoying, but audible, sound to play so that when I am in the other room I know I need to go back to my PC. I have previously used the PC speaker to make a very clear sound for when I was running a very long ansible deployment - with different sounds for if the deployment failed or succeeded, and a shorter one to tell me that a particular stage in the process has passed. So point being, I assume the vast majority of soundpacks would include a light, inoffensive, gentle sound for someone using them all the time, possibly with headphones on. I specifically would benefit from a very audible short piece of music I can hear over the TV whilst I'm washing up in the kitchen or starting to wind down for bed so I know I've hit a breakpoint after like 45 minutes of number-crunching and need to go back to my screen to see what needs done next.

Whilst making packs like themes makes sense, I think users should still be able to override this with just a local path, and I assume this would not be particularly difficult to implement.

**Comment by AFre100** [comment]
@stellarpower In the short term, have you had any success in adjusting the global volume of audio cues in VS Code so they can be more audible than any potential music, other sounds (lowering music player volume), and adjusting speaker volume for other room?

**Comment by stellarpower** [comment]
I have thanks, it's a bit of a more audible bonk (no pun intended) right now, but, I think it would still be good to allow customisation for a local file.

**Comment by Zedelghem** [comment]
@meganrogge Sorry for double posting but I am not sure if you follow closed issues. Since the voting on moving it to the backlog started in the duplicate thread, should we vote here? What is the deadline then?

**Comment by meganrogge** [comment]
Accessibility issues work differently - no need to vote

**Comment by juancampa** [comment]
> Shouldn't we start with overridable audio cues first though? (the same as users can override specific theme colors)

☝️ this was on point. That'll cover the most of the issues I've heard about sound cues. Sound themes could have still have the problem of "what if the user likes some of the sounds in a theme but not others?"
